### ISSUE TITLE
Deprecate jQuery?

### ISSUE THREAD

**Issue by bradfrost** [pattern-lab/styleguidekit-assets-default#78]
This is part of a larger effort refactoring of Pattern Lab's dependencies, but currently jQuery is being used intermittently throughout the codebase. This is more a question than anything, but is it possible to remove that jQuery dependency? My gut says yes, but it would be be great to get a more JS-savvy developer to review.

**Comment by bmuenzenmeyer** [comment]
All of these FE dependencies have a jQuery reference. Not to say it's impossible - just trying to understand the footprint.

* jwerty
* jQueryCookie
* typeahead

**Comment by sghoweri** [comment]
Yeah... jQuery really shouldn't be needed given what the relatively simple JS requirements of Pattern Lab's Styleguidekit. There's plenty of options available w/o a jQuery dependency that should work just as good if not better!

- https://www.npmjs.com/package/mousetrap or https://www.npmjs.com/package/keyboardjs as a jWerty alternative 
- https://www.npmjs.com/package/js-cookie as a jQuery Cookie alternative
- And https://joshuajohnson.co.uk/Choices/, https://goodies.pixabay.com/javascript/auto-complete/demo.html or http://leaverou.github.io/awesomplete/#advanced-examples as a replacement to typeahead.js

**Comment by sghoweri** [comment]
Alternatively, if the navigation was a vertical panel on the side with search at the very top OR if the nav items were horizontal (ala InVision's interface) we could always just combine fuzzy search with filtering the nav items that match up...  http://listjs.com/examples/fuzzy-search/

**Comment by bradfrost** [comment]
@sghoweri thanks for the roundup of non-jQuery options. I'd definitely be on board with migrating away from jQuery to complete those features.

> Alternatively, if the navigation was a vertical panel on the side with search at the very top OR if the nav items were horizontal (ala InVision's interface) we could always just combine fuzzy search with filtering the nav items that match up... http://listjs.com/examples/fuzzy-search/

Interesting concept! Let's park that thought for now as I think there might be some other things we might do with search beyond just pattern navigation in the future. But love all these great ideas!

**Comment by bradfrost** [comment]
TODO: turn these non-jQuery suggestions into actionable tasks
